### PR TITLE
cdc: make mq encoding protocol related more clear

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [x] v8.3 (TiDB 8.3 versions)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/18783
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-avro-protocol.md [hunk 1]
```diff
@@ -272,14 +279,27 @@ DECIMAL(10, 4)
...
+TiCDC Avro protocol support decoding by the [`io.confluent.kafka.serializers.KafkaAvroDeserializer`](https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/serdes-avro.html#avro-deserializer).
+
+The consumer program can acquire the schema by using [Schema Registry API](https://docs.confluent.io/platform/current/schema-registry/develop/api.html), which can be used to deserialize the TiCDC avro protocol encoded data.
+
+### Distinguish event type
+
+Consumer can distinguish DML event type by the following rule:
+
+* If only have Key part, it is a Delete event.
+* If have both Key and Value part, it is a Insert or Update event. If enable the [TiDB extension fields](#tidb-extension-fields), use the `_tidb_op` field to know whether it is Insert or Update, else cannot distinguish these two.
```
Comment (hfxsd) [inline ticdc/ticdc-avro-protocol.md:301]:
```suggestion
### Distinguish event types

The consumer program can distinguish DML event types by the following rules:

* If there is only the Key part, then it is a Delete event.
* If there are both Key and Value parts, then it is either an Insert or an Update event. If the [TiDB extension fields](#tidb-extension-fields) are enabled, you can use the `_tidb_op` field to identify whether it is Insert or Update.  If the TiDB extension fields are not enabled, you cannot distinguish them.
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 2]
```diff
@@ -114,7 +114,13 @@ The following are descriptions of sink URI parameters and values that can be con
...
 > **Note:**
 >
-> When `protocol` is `open-protocol`, TiCDC tries to avoid generating messages that exceed `max-message-bytes` in length. However, if a row is so large that a single change alone exceeds `max-message-bytes` in length, to avoid silent failure, TiCDC tries to output this message and prints a warning in the log.
+> When `protocol` is `open-protocol`, TiCDC tries to batch multiple encoded messages into one Kafka message, avoid generating messages that exceed `max-message-bytes` in length. 
+> If the encoding result of a single row changed event is larger than the `max-message-bytes` in length, changefeed report error, and print logs.
+
+> **注意：**
+>
+> 当 `protocol` 为 `open-protocol` 时，TiCDC 会将多个事件编码到同一个 Kafka 消息中，此过程中尽量避免产生长度超过 `max-message-bytes` 的消息。
+> 如果单条数据变更编码得到的消息大小超过了 `max-message-bytes` 个字节，changefeed 会报错，打印错误日志。
```
Comment (hfxsd) [inline ticdc/ticdc-sink-to-kafka.md:123]:
```suggestion
> When `protocol` is `open-protocol`, TiCDC encodes multiple events into one Kafka message. Avoid generating messages that exceed the length specified by `max-message-bytes`. 
> If the encoded result of a single row change event exceeds the value of `max-message-bytes`, the changefeed reports an error, and prints logs.
```

File: ticdc/ticdc-avro-protocol.md [hunk 3]
```diff
@@ -95,6 +75,34 @@ The `fields` in the key contains only primary key columns or unique index column
 
 The data format of Value is the same as that of Key, by default. However, `fields` in the Value contains all columns, not just the primary key columns.
 
+> **Note:**
+>
+> The Avro protocol encodes DML events as follows: 
+> - For Delete events, only encodes the Key part. The Value part is empty.
```
Comment (hfxsd) [inline ticdc/ticdc-avro-protocol.md:81]:
```suggestion
> The Avro protocol encodes DML events as follows: 
> 
> - For Delete events, only encodes the Key part. The Value part is empty. 
```

File: ticdc/ticdc-avro-protocol.md [hunk 4]
```diff
@@ -5,48 +5,28 @@ summary: Learn the concept of TiCDC Avro Protocol and how to use it.
 
 # TiCDC Avro Protocol
 
-Avro is a data exchange format protocol defined by [Apache Avro™](https://avro.apache.org/) and chosen by [Confluent Platform](https://docs.confluent.io/platform/current/platform.html) as the default data exchange format. This document describes the implementation of the Avro data format in TiCDC, including TiDB extension fields, definition of the Avro data format, and the interaction between Avro and [Confluent Schema Registry](https://docs.confluent.io/platform/current/schema-registry/index.html).
+TiCDC Avro protocol is a third part implementation of the [Confluent Avro](https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/serdes-avro.html) data exchange protocol which is defined by [Confluent Platform](https://docs.confluent.io/platform/current/platform.html). Avro is a data format defined by [Apache Avro™](https://avro.apache.org/)
```
Comment (qiancai) [inline ticdc/ticdc-avro-protocol.md:8]:
```suggestion
TiCDC Avro protocol is a third-party implementation of the [Confluent Avro](https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/serdes-avro.html) data exchange protocol defined by [Confluent Platform](https://docs.confluent.io/platform/current/platform.html). Avro is a data format defined by [Apache Avro™](https://avro.apache.org/)
```

File: ticdc/ticdc-avro-protocol.md [hunk 5]
```diff
@@ -5,48 +5,28 @@ summary: Learn the concept of TiCDC Avro Protocol and how to use it.
...
 ```
-
-```shell
-[sink]
-dispatchers = [
- {matcher = ['*.*'], topic = "tidb_{schema}_{table}"},
-]
-```
-
 The value of `--schema-registry` supports the `https` protocol and `username:password` authentication, for example, `--schema-registry=https://username:password@schema-registry-uri.com`. The username and password must be URL-encoded.
```
Comment (qiancai) [inline ticdc/ticdc-avro-protocol.md:25]:
```suggestion
The value of `--schema-registry` supports the `https` protocol and `username:password` authentication. The username and password must be URL-encoded. For example, `--schema-registry=https://username:password@schema-registry-uri.com`.
```

File: ticdc/ticdc-avro-protocol.md [hunk 6]
```diff
@@ -5,48 +5,28 @@ summary: Learn the concept of TiCDC Avro Protocol and how to use it.
...
-The following is a configuration example:
-
-{{< copyable "shell-regular" >}}
-
-```shell
-cdc cli changefeed create --server=http://127.0.0.1:8300 --changefeed-id="kafka-avro-enable-extension" --sink-uri="kafka://127.0.0.1:9092/topic-name?protocol=avro&enable-tidb-extension=true" --schema-registry=http://127.0.0.1:8081 --config changefeed_config.toml
-```
+> **Note：**
+> 
+> When using the Avro protocol, one Kafka Topic can only have data for one table. You need to configure the [Topic dispatcher](/ticdc/ticdc-sink-to-kafka.md#topic-dispatchers) in the configuration file.
```
Comment (qiancai) [inline ticdc/ticdc-avro-protocol.md:29]:
```suggestion
> When using the Avro protocol, one Kafka topic can only contain data for one table. You need to configure the [Topic dispatcher](/ticdc/ticdc-sink-to-kafka.md#topic-dispatchers) in the configuration file. 
```

File: ticdc/ticdc-avro-protocol.md [hunk 7]
```diff
@@ -95,6 +75,35 @@ The `fields` in the key contains only primary key columns or unique index column
 
 The data format of Value is the same as that of Key, by default. However, `fields` in the Value contains all columns, not just the primary key columns.
```
Comment (qiancai) [inline ticdc/ticdc-avro-protocol.md:76]:
```suggestion
The data format of Value is the same as that of Key, by default. However, `fields` in the Value contains all columns.
```

Comment (qiancai) [reply ticdc/ticdc-avro-protocol.md:76]: 中文 PR 里删除了 “而不仅仅是主键列”

File: ticdc/ticdc-avro-protocol.md [hunk 8]
```diff
@@ -95,6 +75,35 @@ The `fields` in the key contains only primary key columns or unique index column
...
 
+> **Note:**
+>
+> The Avro protocol encodes DML events as follows: 
+> 
+> - For Delete events, only encodes the Key part. The Value part is empty. 
+> - For Insert events, encodes all column data to the Value part. 
+> - For Update events, encodes only all column data that is updated.
```
Comment (qiancai) [inline ticdc/ticdc-avro-protocol.md:84]:
```suggestion
> The Avro protocol encodes DML events as follows: 
> 
> - For Delete events, Avro only encodes the Key part. The Value part is empty. 
> - For Insert events, Avro encodes all column data to the Value part. 
> - For Update events, Avro encodes only all column data that is updated to the Value part.
```

File: ticdc/ticdc-avro-protocol.md [hunk 9]
```diff
@@ -272,14 +281,27 @@ DECIMAL(10, 4)
...
 
-Note that, even if a schema change passes the compatibility check and a new version is registered, the data producers and consumers still need to perform an upgrade to ensure normal running of the system.
+For example, the default [compatibility policy](https://docs.confluent.io/platform/current/schema-registry/fundamentals/schema-evolution.html#compatibility-types) of Confluent Schema Registry is `BACKWARD`, and you add a non-empty column to the source table. In this situation, Avro generates a new schema but fails to register it with Schema Registry due to compatibility issues. At this time, the changefeed enters an error state.
 
-Assume that the default compatibility policy of Confluent Schema Registry is `BACKWARD` and add a non-empty column to the source table. In this situation, Avro generates a new schema but fails to register it with Schema Registry due to compatibility issues. At this time, the changefeed enters an error state.
+Note that, even if a schema change passes the compatibility check and a new version is registered, the data producers and consumers still need to acquire the schema to support data encoding and decoding.
```
Comment (qiancai) [inline ticdc/ticdc-avro-protocol.md:288]:
```suggestion
Note that, even if a schema change passes the compatibility check and a new version is registered, the data producers and consumers still need to acquire the new schema for data encoding and decoding.
```

File: ticdc/ticdc-avro-protocol.md [hunk 10]
```diff
@@ -272,14 +281,27 @@ DECIMAL(10, 4)
...
 
 For more information about schemas, refer to [Schema Registry related documents](https://docs.confluent.io/platform/current/schema-registry/avro.html).
 
+## Consumer implementation
+
+The TiCDC Avro protocol can be deserialized by [`io.confluent.kafka.serializers.KafkaAvroDeserializer`](https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/serdes-avro.html#avro-deserializer).
+
+The consumer program can acquire the schema by using [Schema Registry API](https://docs.confluent.io/platform/current/schema-registry/develop/api.html), and then uses it to deserialize the data encoded by the TiCDC Avro protocol.
```
Comment (qiancai) [inline ticdc/ticdc-avro-protocol.md:296]:
```suggestion
The consumer program can acquire the latest schema via [Schema Registry API](https://docs.confluent.io/platform/current/schema-registry/develop/api.html), and then use the schema to deserialize the data encoded by the TiCDC Avro protocol.
```

File: ticdc/ticdc-avro-protocol.md [hunk 11]
```diff
@@ -272,14 +281,27 @@ DECIMAL(10, 4)
...
+
+### Distinguish event types
+
+The consumer program can distinguish DML event types by the following rules:
+
+* If there is only the Key part, then it is a Delete event.
+* If there are both Key and Value parts, then it is either an Insert or an Update event. If the [TiDB extension fields](#tidb-extension-fields) are enabled, you can use the `_tidb_op` field to identify if it is an Insert or Update event. If the TiDB extension fields are not enabled, you cannot distinguish them.
```
Comment (qiancai) [inline ticdc/ticdc-avro-protocol.md:303]:
```suggestion
* If there is only the Key part, it is a Delete event.
* If there are both Key and Value parts, it is either an Insert or an Update event. If the [TiDB extension fields](#tidb-extension-fields) are enabled, you can use the `_tidb_op` field to identify if it is an Insert or Update event. If the TiDB extension fields are not enabled, you cannot distinguish them.
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 12]
```diff
@@ -80,7 +80,7 @@ The following are descriptions of sink URI parameters and values that can be con
...
 | `compression` | The compression algorithm used when sending messages (value options are `none`, `lz4`, `gzip`, `snappy`, and `zstd`; `none` by default). Note that the Snappy compressed file must be in the [official Snappy format](https://github.com/google/snappy). Other variants of Snappy compression are not supported.|
-| `protocol` | The protocol with which messages are output to Kafka. The value options are `canal-json`, `open-protocol`, `avro`, `debezium`, and `simple`.   |
+| `protocol` | The protocol with which messages are output to Kafka. The value options are [`canal-json`](/ticdc/ticdc-canal-json.md)、[`open-protocol`](/ticdc/ticdc-open-protocol.md)、[`avro`](/ticdc/ticdc-avro-protocol.md)、[`debezium`](/ticdc/ticdc-debezium.md) and [`simple`](/ticdc/ticdc-simple-protocol.md).   |
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-kafka.md:83]: 中文 PR 里 https://github.com/pingcap/docs-cn/pull/18783/files, protocol 这行在 topic-name 参数下面，要统一下位置吗

File: ticdc/ticdc-sink-to-kafka.md [hunk 13]
```diff
@@ -114,7 +114,8 @@ The following are descriptions of sink URI parameters and values that can be con
 
 > **Note:**
 >
-> When `protocol` is `open-protocol`, TiCDC tries to avoid generating messages that exceed `max-message-bytes` in length. However, if a row is so large that a single change alone exceeds `max-message-bytes` in length, to avoid silent failure, TiCDC tries to output this message and prints a warning in the log.
+> When `protocol` is `open-protocol`, TiCDC encodes multiple events into one Kafka message. Avoid generating messages that exceed the length specified by `max-message-bytes`.
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-kafka.md:117]:
```suggestion
> When `protocol` is `open-protocol`, TiCDC encodes multiple events into one Kafka message and avoids generating messages that exceed the length specified by `max-message-bytes`. 
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 14]
```diff
@@ -114,7 +114,8 @@ The following are descriptions of sink URI parameters and values that can be con
 
 > **Note:**
 >
-> When `protocol` is `open-protocol`, TiCDC tries to avoid generating messages that exceed `max-message-bytes` in length. However, if a row is so large that a single change alone exceeds `max-message-bytes` in length, to avoid silent failure, TiCDC tries to output this message and prints a warning in the log.
+> When `protocol` is `open-protocol`, TiCDC encodes multiple events into one Kafka message. Avoid generating messages that exceed the length specified by `max-message-bytes`. 
+> If the encoded result of a single row change event exceeds the value of `max-message-bytes`, the changefeed reports an error, and prints logs.
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-kafka.md:118]:
```suggestion
> If the encoded result of a single row change event exceeds the value of `max-message-bytes`, the changefeed reports an error and prints logs.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Review by hfxsd — APPROVED** [review]

File: ticdc/ticdc-avro-protocol.md [hunk 15]
```diff
@@ -5,48 +5,28 @@ summary: Learn the concept of TiCDC Avro Protocol and how to use it.
...
-When using Message Queue (MQ) as a downstream sink, you can specify Avro in `sink-uri`. TiCDC captures TiDB DML events, creates Avro messages from these events, and sends the messages downstream. When Avro detects a schema change, it registers the latest schema with Schema Registry.
-
 The following is a configuration example using Avro:
 
 {{< copyable "shell-regular" >}}
 
 ```shell
 cdc cli changefeed create --server=http://127.0.0.1:8300 --changefeed-id="kafka-avro" --sink-uri="kafka://127.0.0.1:9092/topic-name?protocol=avro" --schema-registry=http://127.0.0.1:8081 --config changefeed_config.toml
 ```
+The value of `--schema-registry` supports the `https` protocol and `username:password` authentication. The username and password must be URL-encoded. For example, `--schema-registry=https://username:password@schema-registry-uri.com`.
```
Comment (hfxsd) [inline ticdc/ticdc-avro-protocol.md:25]:
```suggestion

The value of `--schema-registry` supports the `https` protocol and `username:password` authentication. The username and password must be URL-encoded. For example, `--schema-registry=https://username:password@schema-registry-uri.com`.
```

**Pull request merged**